drivers/mmcsd/mmcsd_spi.c

```diff
@@ -61,7 +61,9 @@
 #  define CONFIG_MMCSD_NSLOTS 1
 #endif
 
-#define MMCSD_IDMODE_CLOCK           (400000)
+#ifndef CONFIG_MMCSD_IDMODE_CLOCK
+#  define CONFIG_MMCSD_IDMODE_CLOCK 400000
+#endif
```

Comment:
I think there's no need for this, Kconfig will always define it if you have the default

Reply:
This is a pattern I see in a lot of places, for example right above with `CONFIG_MMCSD_NSLOTS` so I keeping it.  Should I remove that as well?

Reply:
Yeah, I would say it is best to remove it

Reply:
> 
> 
> This is a pattern I see in a lot of places, for example right above with `CONFIG_MMCSD_NSLOTS` so I keeping it. Should I remove that as well?

It is a holdover from the days before kconfig-frontends when .config files were edited by hand.  In that case, extensive verification of configuration settings was needed at compile time.  But it is probably no longer necessary; we depend on the Kconfigs to assure that everything is defined and valid.

There is a lot of old stuff in config.h generated in mkconfig.c that could be removed too.

Reply:
@v01d I removed the config defaults in that file that are defined in `Kconfig`.  There is one more config  `CONFIG_MMCSD_SECTOR512` but it does not look to be in Kconfig so I don't want to touch it.